src/vm/functions/conversions.rs

```diff
@@ -113,3 +115,107 @@ pub fn native_buff_to_uint_be(value: Value) -> Result<Value> {
     }
     return buff_to_int_generic(value, EndianDirection::BigEndian, convert_to_uint_be);
 }
+
+pub fn native_string_to_int_generic(
+    value: Value,
+    conversion_fn: fn(String) -> Result<Value>,
+) -> Result<Value> {
+    match value {
+        Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData { data }))) => {
+            let as_string = String::from_utf8(data).unwrap();
+            return conversion_fn(as_string);
+        }
+        Value::Sequence(SequenceData::String(CharType::UTF8(UTF8Data { data }))) => {
+            let flat = data.into_iter().flatten().collect();
+            let as_string = String::from_utf8(flat).unwrap();
+            return conversion_fn(as_string);
+        }
+        _ => {
+            return Err(CheckErrors::UnionTypeValueError(
+                vec![
+                    TypeSignature::max_string_ascii(),
+                    TypeSignature::max_string_utf8(),
+                ],
+                value,
+            )
+            .into())
+        }
+    };
+}
+
+pub fn native_string_to_int(value: Value) -> Result<Value> {
+    fn convert_to_int(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<i128>();
+        match possible_int {
+            Ok(val) => return Ok(Value::Int(val)),
+            Err(error) => return Err(CheckErrors::InvalidCharactersDetected.into()),
+        }
+    }
+    return native_string_to_int_generic(value, convert_to_int);
+}
+
+pub fn native_string_to_uint(value: Value) -> Result<Value> {
+    fn convert_to_uint(raw_string: String) -> Result<Value> {
+        let possible_int = raw_string.parse::<u128>();
+        match possible_int {
+            Ok(val) => return Ok(Value::UInt(val)),
+            Err(error) => return Err(CheckErrors::InvalidCharactersDetected.into()),
```

Comment:
Same as with `natve_string_to_int()`, this *should not abort the transaction* if the user submits a string that is too big (or small) to be represented in a `i128` or `u128`.  This should be a runtime error that the caller can handle, so the return type here should be a `(optional int|uint)` or `(response int|uint int)`.

Reply:
yes.. it's an optional now